vcd/resource_vcd_org_user.go

```diff
@@ -0,0 +1,351 @@
+package vcd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+func resourceVcdOrgUser() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdOrgUserCreate,
+		Read:   resourceVcdOrgUserRead,
+		Delete: resourceVcdOrgUserDelete,
+		Update: resourceVcdOrgUserUpdate,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdOrgUserImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCase("lower"),
+				Description:  "User's name. Only lowercase letters allowed. Cannot be changed after creation",
+			},
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Organization this user belongs to",
+			},
+			"role": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    false,
+				Description: "Role within the organization",
+			},
+			"password": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: false,
+
+				ConflictsWith: []string{"password_file"},
+				Description: "The user's password. This value is never returned on read. " +
+					"It is inspected on create and modify. " +
+					"On modify, the absence of this element indicates that the password should not be changed.",
+			},
+			"password_file": &schema.Schema{
```

Comment:
One thing I see about the password file is that it is hard to change the password (you must change file name).

Don't know how to make it better other than state including timestamp of when the file was modified  in a separate Computed field and comparing against it. Up to you if it's worth it :)

Reply:
I have thought of adding a "change_password" Boolean field for this purpose.
Both approaches ("password" and "password_file") have pros and cons. For automation, I think using a password file that can be generated and removed in a script is the easy approach.

The *easiest* approach would be to update using the "password" field but then you would have to apply again with an empty password to remove it from the state file.